src/command/backup/blockIncr.h

```diff
@@ -0,0 +1,46 @@
+/***********************************************************************************************************************************
+Block Incremental Filter
+
+The block incremental filter builds a block list and map (see BlockMap object) either from a file (the first time) or from a file
+and a prior map (each subsequent time). The block list is stored first and the block map second. The filter returns the size of the
+block map so it can be extracted separately.
+
+The map is duplicated in each backup where the file has changed so the block map only needs to be retrieved from the most recent
+backup. However, the block map may reference block lists (or parts thereof) in any prior (or the current) backup.
+
+The block list consists of a series of blocks stored as chunks (see IoChunk filter) to make the format more flexible. Each block
+consists of the following, compressed and encrypted as required:
+
+- A varint-128 encoded block number stored as a delta of the previous block. So, if the first block is 138 it would be stored as 138
```

Comment:
Clarifying how sequential reads work, specifically how storing the block number at the beginning of the block provides enough information to scan a compressed file.

It looks like:
- the original blocks were fixed size cleartext, so we just keep reading cleartext up to that size,
- the IoFilters stop cleanly after providing the fixed size data.

These both seem like reasonable assumptions. Is that what you have in mind?

Reply:
The chunk headers provide information about the block size as stored, so we just read to the end of chunk marker. The block no is used to decide where the block goes in the destination file but is not used for decoding the block.